### PR TITLE
LPD-38542 Fix Issue in DDMServiceUpgradeStepRegistrator Due to Incorrect Upgrade Backport

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-38542](https://liferay.atlassian.net/browse/LPD-38542)

This issue was caused by a [backport](https://liferay.atlassian.net/browse/BPR-77967) where a dummy [upgrade](https://github.com/liferay/liferay-portal-ee/blob/2024.q2.12/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/registry/DDMServiceUpgradeStepRegistrator.java#L568) was added.  As a result, anyone upgrading from this tag to a later release will skip the necessary [upgrade](https://github.com/liferay/liferay-portal-ee/blob/2024.q3.3/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/registry/DDMServiceUpgradeStepRegistrator.java#L570-L579).

[LPD-38542]: https://liferay.atlassian.net/browse/LPD-38542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by marcelabc** [comment]
Just started reviewing :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/registry/DDMServiceUpgradeStepRegistrator.java [hunk 1]
```diff
@@ -570,8 +570,18 @@ protected String[] getTableNames() {
...
 
+		registry.register("5.4.5", "5.5.0", new DummyUpgradeStep());
+
+		registry.register(
+			"5.5.0", "5.5.1",
+			new com.liferay.dynamic.data.mapping.internal.upgrade.v5_5_1.
+				DDMFieldUpgradeProcess(),
+			new DDMFieldAttributeUpgradeProcess());
+
+		registry.register("5.5.1", "5.5.2", new DummyUpgradeStep());
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/registry/DDMServiceUpgradeStepRegistrator.java:581]: As we discussed, remove this dummy upgrade.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/registry/DDMServiceUpgradeStepRegistrator.java [hunk 2]
```diff
@@ -570,8 +570,18 @@ protected String[] getTableNames() {
...
+
 		registry.register(
-			"5.4.5", "5.5.0",
+			"5.5.2", "5.6.0",
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/registry/DDMServiceUpgradeStepRegistrator.java:584]: Replace `5.5.2` to `5.5.1`

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:upgrade

**Comment by pedro-oliveira446** [comment]
ci:test:forms

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:upgrade

**Comment by pedro-oliveira446** [comment]
ci:test:forms

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Pull request closed without merging**